db/db_test.cc

```diff
@@ -1352,10 +1356,28 @@ TEST_F(DBTest, ApproximateSizesMemTable) {
   start = Key(1050);
   end = Key(1080);
   r = Range(start, end);
-  db_->GetApproximateSizes(&r, 1, &size_with_mt, include_both);
+  db_->GetApproximateSizes(sizeApproxOpt, defaultColumnFamily, &r, 1,
+                           &size_with_mt);
   db_->GetApproximateSizes(&r, 1, &size_without_mt);
   ASSERT_GT(size_with_mt, size_without_mt);
   ASSERT_GT(size_without_mt, 6000);
+
+  // Check that include_memtabtles flag works as expected
+  sizeApproxOpt.include_memtabtles = false;
+  db_->GetApproximateSizes(sizeApproxOpt, defaultColumnFamily, &r, 1, &size);
+  ASSERT_EQ(size, size_without_mt);
+
+  // Check that full_to_intersecting_file_size_ratio works as expected, when
+  // the heurostic conditions are not met
```

Comment:
You are going to add a test case where the condition is met, right?

Reply:
@siying I am actually not sure how to create the desired structure of files in the different level. Please advise.

Reply:
Several ways you can do to achieve it (unfortunately none is very easy). Let me try to find an example test case doing it for you.

Reply:
I think about it, the best way might be to first create a bottom-most level, and then insert some keys and expect RocksDB can naturally fill some data in other levels.
First insert keys you want to show up in the last level, and call CompactRange() to force the keys there.
And then insert keys that will go to other levels. I can explain it in person.